### PR TITLE
Fetch token prices via wallet api v2

### PR DESCRIPTION
**Pull request by Perronef5**

### CODE REVIEW

File: src/components/TokenPricesTicker.tsx [hunk 1]
```diff
@@ -21,21 +22,27 @@ const TokenPricesTicker = ({ ...boxProps }: Props) => {
...
-  )
+  const [triggerGetTokenPrices, { isFetching, data: tokenPrices }] =
+    useLazyGetTokenPricesQuery({
+      pollingInterval: 1000 * 60,
+      refetchOnReconnect: true,
+      refetchOnFocus: true,
+    })
+
+  useAsync(async () => {
+    await triggerGetTokenPrices({ tokens: 'helium,solana', currency }, false)
```
Comment (tyler-whitman) [inline src/components/TokenPricesTicker.tsx:33]: So this triggers it to keep polling? Does it automatically stop when the page loses focus? Or do we want it to keep running always when the app is open, I think that might still be ok?

File: src/features/account/AccountTokenCurrencyBalance.tsx [hunk 2]
```diff
@@ -44,11 +45,14 @@ const AccountTokenCurrencyBalance = ({
 
     switch (ticker) {
       case 'HNT':
-        if (staked) {
-          toCurrencyString(networkStakedBalance).then(setBalanceString)
-        } else {
-          toCurrencyString(networkBalance).then(setBalanceString)
-        }
+        toCurrencyString(
+          networkStakedBalance
```
Comment (tyler-whitman) [inline src/features/account/AccountTokenCurrencyBalance.tsx:49]: Should this be conditional on `staked` like before? Or ok to use `networkStakedBalance`?

File: src/utils/Balance.tsx [hunk 3]
```diff
@@ -367,35 +366,50 @@ const useBalanceHook = () => {
...
+      balance?: Balance<NetworkTokens | TestNetworkTokens>,
+      ticker: Ticker = 'HNT',
+    ): Promise<string> => {
       const defaultResponse = new Promise<string>((resolve) => resolve(''))
-      if (!balance || balance?.floatBalance === undefined) {
-        return defaultResponse
+
+      let bal = new Balance(
+        0,
+        ticker === 'SOL' ? CurrencyType.solTokens : CurrencyType.networkToken,
```
Comment (tyler-whitman) [inline src/utils/Balance.tsx:381]: Do we have any const we can read from for `SOL` and `HNT`?

**Review by tyler-whitman — APPROVED** [review]

File: src/components/TokenPricesTicker.tsx (hunk 1 above)
Comment (Perronef5) [reply src/components/TokenPricesTicker.tsx:33]: The second argument is used to tell redux if you want to keep the cache or not. Polling is set to 1 minute on the hook in this file

File: src/features/account/AccountTokenCurrencyBalance.tsx (hunk 2 above)
Comment (Perronef5) [reply src/features/account/AccountTokenCurrencyBalance.tsx:49]: I think if you have a `networkStakedBalance` then we should add it no?

File: src/utils/Balance.tsx (hunk 3 above)
Comment (Perronef5) [reply src/utils/Balance.tsx:381]: We do let me use `TICKER`

File: src/features/account/AccountTokenCurrencyBalance.tsx (hunk 2 above)
Comment (tyler-whitman) [reply src/features/account/AccountTokenCurrencyBalance.tsx:49]: Yeah that makes sense, not sure what the staked boolean was checking before

File: src/utils/Balance.tsx (hunk 3 above)
Comment (matthewcarlreetz) [reply src/utils/Balance.tsx:381]: Ticker is a discriminated union, so should be good. You could also use `Balance.fromIntAndTicker(0, ticker)`

**Pull request merged**